### DISCUSSION TITLE
Creating a clean system

### DISCUSSION THREAD

**Discussion by SirNibble** [prirun/p50em#14] [General]
I love playing with my Rev20 system, but it's based on the example server and patch-worked from the others for such things as Iinformation & Prime Pascal, and tools like 'up','pwd' and 'tree'. 

It works, but doesn't feel 'substainable' as I've not documented my patchwork and there's no distribution tapes that I've found. This also hinders any support as there's not a known clean-sheet to start from.

My usage is mainly redisovering . . .

   Information
   Prime Pascal (SEG & EPF)
   PLP/PL1
   Emacs
   Primenet
   Batch & Phantom jobs
   Spooler - if it can spool to the host file system by some means
   
Being able to delve into stuff like the FORTRAN/F77 & COBOL compilers, and to discover what the fuss was all about with MIDAS, would be great, but not an immediate use. I never got into the nitty gritty of PMA or RPG etc, so that would be fun at some point.

So given the Distribution Tapes we have, and the copies of the manuals we have, what would be the 'sweet spot' for a clean installation?

I'm erring toward Rev21 - it's the latest revision I remember using and there seems to be more manuals for it.  

(I also tend to use -cpuid 9955ii as it's what we had, if that helps.)

Thoughts?

**Comment by drboone** [comment]
The cpu model won't make much difference.  I don't recall seeing any Prime products that enforced a cpu-class license.  I think the Rev18 example won't boot on some of the later CPU types, but it's not like we have even a complete base PRIMOS at that rev.

My personal taste would be to do a 19.4 or 21.x system, if I had the os and layered products.  (I don't.)  I've been using a 23.3 machine as my workhorse.

If you want to have proper dist tapes, then the best option would eventually be that 23.3 set I have, but INFORMATION isn't there.

I'd suggest a PRIMENET ring, where you have INFORMATION on a 21.x machine installed from those tapes, and a 23.3 machine with the compilers and such, but you'd have to fudge PRIMENET on 21.x.  Or perhaps see if the INFORMATION on the 21.x tapes will install to a 23.x system.

I think EMACS will install across a number of os versions, and we have 19.4 and 23.3 (https://yagi.h-net.org/prime_software/), so if it isn't in the set you want to use, there's still a manageable option.  I have a number of versions of the INDEX / T_BUILD_TOOLS software I can share, if they're not on the set you choose, so PLP/SPL/DEREMER aren't a problem.  CBL, CC, DBG, F77, MIDASPLUS, PASCAL, PL1 are on the 23.3 set.  I think the only form in which I have RPG is mixed into installed systems, but I might be able to reconstruct an install kit for them.  Looks like two versions, one unlabeled, the other 19.4ish.  PMA is in the base os, of course.

If it's any consolation, it was pretty common to have mixed-version installs back then.  At least by installing from the dist tape sets, you can write a procedure for re-creating the thing.

I should probably add a description of the dist tape sets that are up for download.  They've been stripped of serialization and re-saved so that each logical is a single volume.  A real dist tape set tended to run to multiple reels, and the logical saves were just packed on until end of reel, then spanned onto the next.  That is, you'd get Boot, U1, part of U2, next reel, rest of U2, C1, ..., next reel, IC, KLMTF, sort of thing.  Since the emulator doesn't have a facility for limiting tape size, that's a hard thing to produce.

I'll push on that 23.3 set, it's time.

Wall of text, sorry.

**Comment by SirNibble** [comment]
Given the amount of things in the R23.3 you've just put up, R23.3 seems a grand choice!

Thanks for that.
